test/builtin_dataset_mocks.py

```diff
@@ -1431,3 +1431,48 @@ def stanford_cars(info, root, config):
         make_tar(root, "car_devkit.tgz", devkit, compression="gz")
 
     return num_samples
+
+
+@register_mock
+def sun397(info, root, config):
+    images_root = root / "SUN397"
+    images_root.mkdir()
+
+    categories = ["abbey", "airplane_cabin", "wrestling_ring/indoor"]
+    category_keys = [f"/{category[0]}/{category}" for category in categories]
+
+    keys = []
+    for key in category_keys:
+        parts = key.split("/")
+        image_files = create_image_folder(
+            root=images_root.joinpath(*parts[1:-1]),
+            name=parts[-1],
+            file_name_fn=lambda idx: f"sun_{idx:05d}.jpg",
+            num_examples=5,
+        )
+
+        keys.extend([f"/{image_file.relative_to(images_root).as_posix()}" for image_file in image_files])
+
+    partitions_root = root / "Partitions"
+    partitions_root.mkdir()
+
+    for fold in range(1, 11):
+        random.shuffle(keys)
+
+        for split, keys_in_split in random_group(keys, ["train", "test"]).items():
+            if split == config.split and str(fold) == config.fold:
+                num_samples = len(keys_in_split)
+
+            with open(partitions_root / f"{split.capitalize()}ing_{fold:02d}.txt", "w") as fh:
+                fh.write("\n".join(sorted(keys_in_split)))
+
+    # Both archives contain this file. Although it shouldn't be used at runtime, it serves as sentinel to test the
+    # filtering of the correct files
```

Comment:
See https://github.com/pytorch/vision/blob/main/torchvision/prototype/datasets/_builtin/README.md#tests for details.